arch/arm64/core/el3_service.S

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023 Chad Karaginides <quic_chadk@quicinc.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This file implements the logic to check that all criteria are met
+ * to enter an EL3 service handler.
+ *
+ * Service available:
+ * Secure Monitor Call (SMC) handler.
+ *
+ * See https://developer.arm.com/docs/den0028/latest
+ */
+
+#include <zephyr/toolchain.h>
+#include <zephyr/linker/sections.h>
+#include <offsets_short.h>
+#include <zephyr/arch/cpu.h>
+
+GTEXT(z_arm64_el3_service)
+SECTION_FUNC(TEXT, z_arm64_el3_service)
+
+	mrs	x0, currentEL
+	lsr     x0, x0, #2
+	cmp	x0, #3         /* Check if in EL3 */
+	bne	1f
+	mrs	x0, esr_el3
+	lsr	x0, x0, #26
+	cmp	x0, #0x17      /* 0x17 = SMC */
+	bne	1f
```

Comment:
We are already demuxing in `z_arm64_sync_exc`, can you move this there instead?

Reply:
Yes I can.  I actually started with the logic there in z_arm64_sync_exc but moved it because I didn't want to clutter z_arm64_sync_exc and that function lives in switch.S which I thought this didn't really fit.  But yes, I will absolutely move it.